src/test/scala/nl/knaw/dans/easy/schemaExamples/SchemaValidationFixture.scala

```diff
@@ -15,19 +15,128 @@
 
-import scala.util.{ Failure, Try }
-import scala.xml.{ Elem, SAXParseException }
+import scala.util.{ Failure, Success, Try }
+import scala.xml._
 
-trait SchemaValidationFixture extends FlatSpec with Matchers {
+trait SchemaValidationFixture extends FlatSpec with Matchers with TableDrivenPropertyChecks {
 
+  val publicSchema: String
+  val localSchemaFile: String
+  val examples: TableFor1[String]
 
+  private val publicEasySchemaBase = "//easy.dans.knaw.nl/schemas"
+  protected val httpsEasySchemaBase = s"https:$publicEasySchemaBase"
+
+  private lazy val triedPublicSchema: Try[Schema] = Try {
+    SchemaFactory
+      .newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI)
+      .newSchema(Array(new StreamSource(publicSchema.toString)).toArray[Source])
+  }
+  private lazy val triedLocalSchema: Try[Schema] = Try {
+    // lazy for two reasons:
+    // - schemaFile is set by concrete test class
+    // - postpone loading until actually validating
+    val xsdInputStream = File(localSchemaFile).contentAsString.replaceAll(
+      s"""schemaLocation="http:$publicEasySchemaBase""",
+      s"""schemaLocation="file://$schemaDir"""
+    ).inputStream
+    SchemaFactory
+      .newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI)
+      .newSchema(Array(new StreamSource(xsdInputStream)).toArray[Source])
+  }
+
+  "examples" should "be schema valid with local copy of easy-schema" in {
+    forEvery(examples) { example =>
+      val xml = XML.loadFile((exampleDir / example).toString())
+      easyLocationsIn(xml).foreach(File(_).toJava should exist)
+      assume(schemaIsOnline(triedLocalSchema)) // inside the loop to not block other errors
+      validate(triedLocalSchema, toLocalSchemas(xml)) shouldBe a[Success[_]]
+    }
+  }
+
+  it should "be schema valid with (unqualified) public schema" in {
+    // sub-schemas may not have an unqualified public schema
+    assume(schemaIsOnline(triedPublicSchema)) // outside the loop as we are only validating
+    assume(lastLocalIsPublic)
+    forEvery(examples) { example =>
+      val xml = XML.loadFile((exampleDir / example).toString())
+      validate(triedPublicSchema, xml.toString) shouldBe a[Success[_]]
+    }
+  }
+
+  private def lastLocalIsPublic: Boolean = {
+    // to ignore tests while easy-schema and easy-schema-examples are under parallel development
+    // and the new schema's are not yet published
+    // reported with: ...lastLocalIsPublic was false
+    val newPublicSchema = localSchemaFile.replace(schemaDir.toString(), httpsEasySchemaBase)
+    Try(new URL(newPublicSchema).openStream()).fold(_ => false, _ => true)
+  }
+
+  private def validate(schema: Try[Schema], xmlString: String): Try[Unit] = {
+    schema.map(_.newValidator().validate(new StreamSource(xmlString.inputStream))) match {
+      case Failure(e: SAXParseException) =>
+        showErrorWithSourceContext(xmlString, e)
+        Failure(e)
+      case x => x
+    }
+  }
+
+  private def easyLocationsIn(xml: Elem): Seq[String] = {
```

Comment:
I've had to spend some time with this function to completely understand what it is doing. May I suggest a little rewrite that hopefully makes it better readable? This includes the 2 comments I made about this function above/below (depending on how GitHub renders things these days 😉)

```scala
private def easyLocationsIn(xml: Elem): Iterable[File] = {
  val attributeValues = xml.attributes.asAttrMap.collect {
    case ("xsi:noNamespaceSchemaLocation", value: String) => s"dummyUri $value"
    case ("xsi:schemaLocation", value: String) => value
  }

  // now we should have strings formatted like: "nsUri1 xsdUrl1 nsUri2 xsdUrl2"
  for {
    value <- attributeValues
    tuples <- value.trim.split(" +").grouped(2) // extract (uri,url) tuples
    url = tuples.applyOrElse(1, "").toString // get second element of each tuple
    if url contains "easy.dans"
  } yield File(url.replace(httpsEasySchemaBase, schemaDir.toString()))
}
```